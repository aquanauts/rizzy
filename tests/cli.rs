use assert_cmd::Command;
use predicates::prelude::*;

#[test]
fn test_timezone_spelling() {
    let assert = Command::cargo_bin("rizzy")
        .unwrap()
        .arg("--zone")
        .arg("America/NewYork")
        .assert();

    assert
        .failure()
        .stderr(predicate::str::contains("America/New_York"));
}

/// Make sure the examples from the README work.
#[test]
fn test_from_readme() {
    let assert = Command::cargo_bin("rizzy")
        .unwrap()
        .arg("--chi")
        .arg("tests/some.log.file")
        .assert();

    let assert = assert.success().stderr(predicate::eq(""));
    let stdout = String::from_utf8(assert.get_output().stdout.clone()).unwrap();
    insta::assert_display_snapshot!(stdout);
}
