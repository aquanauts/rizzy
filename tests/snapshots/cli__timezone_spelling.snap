---
source: tests/cli.rs
assertion_line: 13
expression: stderr

---
[rizzy] Error: Unknown timezone 'America/NewYork', did you mean: ["America/New_York"]

